lib/has_content_id.rb

```diff
@@ -0,0 +1,14 @@
+module HasContentId
+  extend ActiveSupport::Concern
+
+  included do
+    before_validation :generate_content_id, on: :create
+    validates :content_id, presence: true
+  end
+
+private
```

Comment:
Mixing in a private method feels like implementation detail. Perhaps it might be nicer to just use the block form of `before_validation` and dispense with the private helper-style method?